stage0/run.go

```diff
@@ -452,7 +452,7 @@ func prepareStage1Image(cfg PrepareConfig, img types.Hash, cdir string, useOverl
 func setupStage1Image(cfg RunConfig, img types.Hash, cdir string, useOverlay bool) error {
 	if useOverlay {
 		s1 := common.Stage1ImagePath(cdir)
-		if err := overlayRender(cfg, img, cdir, s1); err != nil {
+		if err := overlayRender(cfg, img, cdir, s1, "stage1"); err != nil {
```

Comment:
If we consider a scenario where somebody is smart and has an image `example.com/totally/not/rkt/stage1`, then for this image, the appName variable will hold "stage1". And that conflicts with the hardcoded name for stage1 in this line.

For the real, rkt stage1, consider using a name that is valid directory name, but an invalid acName. Like `^stage1^` or `-stage1-` or just anything that does not match the `^[a-z0-9]+([-][a-z0-9]+)*$` regexp.